<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Create chromR object</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for create.chromR {vcfR}"><tr><td>create.chromR {vcfR}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Create chromR object</h2>

<h3>Description</h3>

<p>Creates and populates an object of class chromR.
</p>


<h3>Usage</h3>

<pre>
create.chromR(vcf, name = "CHROM", seq = NULL, ann = NULL, verbose = TRUE)

vcfR2chromR(x, vcf)

seq2chromR(x, seq = NULL)

ann2chromR(x, gff)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>vcf</code></td>
<td>
<p>an object of class vcfR</p>
</td></tr>
<tr valign="top"><td><code>name</code></td>
<td>
<p>a name for the chromosome (for plotting purposes)</p>
</td></tr>
<tr valign="top"><td><code>seq</code></td>
<td>
<p>a sequence as a DNAbin object</p>
</td></tr>
<tr valign="top"><td><code>ann</code></td>
<td>
<p>an annotation file (gff-like)</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>should verbose output be printed to the console?</p>
</td></tr>
<tr valign="top"><td><code>x</code></td>
<td>
<p>an object of class chromR</p>
</td></tr>
<tr valign="top"><td><code>gff</code></td>
<td>
<p>a data.frame containing annotation data in the gff format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates and names a chromR object from a name, a chromosome (an ape::DNAbin object), variant data (a vcfR object) and annotation data (gff-like).
The function <strong>create.chromR</strong> is a wrapper which calls functions to populate the slots of the chromR object.
</p>
<p>The function <strong>vcf2chromR</strong> is called by create.chromR and transfers the data from the slots of a vcfR object to the slots of a chromR object.
It also tries to extract the 'DP' and 'MQ' fileds (when present) from the fix slot's INFO column.
It is not anticipated that a user would need to use this function directly, but its placed here in case they do.
</p>
<p>The function <strong>seq2chromR</strong> is currently defined as a generic function.
This may change in the future.
This function takes an object of class DNAbin and assigns it to the 'seq' slot of a chromR object.
</p>
<p>The function <strong>ann2chromR</strong> is called by create.chromR and transfers the information from a gff-like object to the 'ann' slot of a chromR object.
It is not anticipated that a user would need to use this function directly, but its placed here in case they do.
</p>


<h3>See Also</h3>

<p><code><a href="../../vcfR/help/chromR-class.html">chromR-class</a></code>,
<code><a href="../../vcfR/help/vcfR-class.html">vcfR-class</a></code>,
<code><a href="../../ape/help/DNAbin.html">DNAbin</a></code>,
<a href="https://github.com/samtools/hts-specs">VCF specification</a>
<a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md">gff3 format</a>
</p>


<h3>Examples</h3>

<pre>
library(vcfR)
data(vcfR_example)
chrom &lt;- create.chromR('sc50', seq=dna, vcf=vcf, ann=gff)
head(chrom)
chrom
plot(chrom)

chrom &lt;- masker(chrom, min_QUAL = 1, min_DP = 300, max_DP = 700, min_MQ = 59, max_MQ = 61)
chrom &lt;- proc.chromR(chrom, win.size=1000)

plot(chrom)
chromoqc(chrom)




</pre>

<hr /><div style="text-align: center;">[Package <em>vcfR</em> version 1.15.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
