<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Convert a vcfR object to MigrateN input file</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for vcfR2migrate {vcfR}"><tr><td>vcfR2migrate {vcfR}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Convert a vcfR object to MigrateN input file</h2>

<h3>Description</h3>

<p>The function converts a vcfR object to a text format that can be used as an infile for MigrateN.
</p>


<h3>Usage</h3>

<pre>
vcfR2migrate(
  vcf,
  pop,
  in_pop,
  out_file = "MigrateN_infile.txt",
  method = c("N", "H")
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>vcf</code></td>
<td>
<p>a vcfR object.</p>
</td></tr>
<tr valign="top"><td><code>pop</code></td>
<td>
<p>factor indicating population membership for each sample.</p>
</td></tr>
<tr valign="top"><td><code>in_pop</code></td>
<td>
<p>vector of population names indicating which population to include in migrate output file.</p>
</td></tr>
<tr valign="top"><td><code>out_file</code></td>
<td>
<p>name of output file.</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>should 'N' or 'H' format data be generated?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function converts a vcfR object to a text file which can be used as input for MigrateN.
The function will remove loci with missing data, indels, and loci that are not bialleleic (loci with more than two alleles). 
Thus, only SNP data analysed where the length of each locus (inmutational steps) is 1 (as opposed to microsatellites or indels).
</p>
<p>The output file should contain Unix line endings (&quot;\n&quot;).
Note that opening the output file in a Windows text editor (just to validate number of markers, individuals or populations) might change the end of line character (eol) to a Windows line ending (&quot;\r\n&quot;). 
This may produce an error running migrate-n.
Because these are typically non-printing characters, this may be a difficult problem to troubleshoot.
The easiest way to circumvent the problem is to transfer the output file to Unix machine and view it there.
If you do introduce Windows line endings you can convert them back to Unix with a program such as 'dos2unix' or 'fromdos' to change the line endings.
</p>


<h3>Value</h3>

<p>a text file that can be used as an input for MigrateN software (SNP format).
</p>


<h3>Author(s)</h3>

<p>Shankar Shakya and Brian J. Knaus
</p>


<h3>See Also</h3>

<p><a href="http://popgen.sc.fsu.edu/Migrate/Migrate-n.html">Migrate-N</a> website.
</p>


<h3>Examples</h3>

<pre>
## Not run: 
data(vcfR_example)
my_pop &lt;- as.factor(paste("pop_", rep(c("A", "B", "C"), each = 6), sep = ""))
vcfR2migrate(vcf = vcf , pop = my_pop , in_pop = c("pop_A","pop_C"),
             out_file = "my2pop.txt", method = 'H')

## End(Not run)


</pre>

<hr /><div style="text-align: center;">[Package <em>vcfR</em> version 1.15.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
