<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Project individuals onto existing principal component axes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsPCASampLoading {SNPRelate}"><tr><td>snpgdsPCASampLoading {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Project individuals onto existing principal component axes
</h2>

<h3>Description</h3>

<p>To calculate the sample eigenvectors using the specified SNP loadings
</p>


<h3>Usage</h3>

<pre>
snpgdsPCASampLoading(loadobj, gdsobj, sample.id=NULL, num.thread=1L,
    verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>loadobj</code></td>
<td>
<p>a <code>snpgdsPCASNPLoadingClass</code> or
<code>snpgdsEigMixSNPLoadingClass</code> object returned from
<a href="../../SNPRelate/help/snpgdsPCASNPLoading.html">snpgdsPCASNPLoading</a></p>
</td></tr>
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>num.thread</code></td>
<td>
<p>the number of CPU cores used</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>sample.id</code> are usually different from the samples used in the
calculation of SNP loadings.
</p>


<h3>Value</h3>

<p>Returns a <code>snpgdsPCAClass</code> object, and it is a list:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>the sample ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>the SNP ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>eigenval</code></td>
<td>
<p>eigenvalues</p>
</td></tr>
<tr valign="top"><td><code>eigenvect</code></td>
<td>
<p>eigenvactors, &ldquo;# of samples&rdquo; x &ldquo;eigen.cnt&rdquo;</p>
</td></tr>
<tr valign="top"><td><code>TraceXTX</code></td>
<td>
<p>the trace of the genetic covariance matrix</p>
</td></tr>
<tr valign="top"><td><code>Bayesian</code></td>
<td>
<p>whether use bayerisan normalization</p>
</td></tr>
</table>
<p>Or returns a <code>snpgdsEigMixClass</code> object, and it is a list:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>the sample ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>the SNP ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>eigenval</code></td>
<td>
<p>eigenvalues</p>
</td></tr>
<tr valign="top"><td><code>eigenvect</code></td>
<td>
<p>eigenvactors, &ldquo;# of samples&rdquo; x &ldquo;eigen.cnt&rdquo;</p>
</td></tr>
<tr valign="top"><td><code>afreq</code></td>
<td>
<p>allele frequencies</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Patterson N, Price AL, Reich D (2006)
Population structure and eigenanalysis. PLoS Genetics 2:e190.
</p>
<p>Zhu, X., Li, S., Cooper, R. S., and Elston, R. C. (2008).
A unified association analysis approach for family and unrelated samples
correcting for stratification. Am J Hum Genet, 82(2), 352-365.
</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsPCA.html">snpgdsPCA</a></code>, <code><a href="../../SNPRelate/help/snpgdsPCACorr.html">snpgdsPCACorr</a></code>,
<code><a href="../../SNPRelate/help/snpgdsPCASNPLoading.html">snpgdsPCASNPLoading</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

sample.id &lt;- read.gdsn(index.gdsn(genofile, "sample.id"))

# first PCA
pca &lt;- snpgdsPCA(genofile, eigen.cnt=8)
snp_load &lt;- snpgdsPCASNPLoading(pca, genofile)

# calculate sample eigenvectors from SNP loadings
samp_load &lt;- snpgdsPCASampLoading(snp_load, genofile, sample.id=sample.id[1:100])

diff &lt;- pca$eigenvect[1:100,] - samp_load$eigenvect
summary(c(diff))
# ~ ZERO


# combine eigenvectors
allpca &lt;- list(
    sample.id = c(pca$sample.id, samp_load$sample.id),
    snp.id = pca$snp.id,
    eigenval = c(pca$eigenval, samp_load$eigenval),
    eigenvect = rbind(pca$eigenvect, samp_load$eigenvect),
    varprop = c(pca$varprop, samp_load$varprop),
    TraceXTX = pca$TraceXTX
)
class(allpca) &lt;- "snpgdsPCAClass"
allpca


# close the genotype file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
