<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Conversion from GDS to PLINK BED</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsGDS2BED {SNPRelate}"><tr><td>snpgdsGDS2BED {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Conversion from GDS to PLINK BED</h2>

<h3>Description</h3>

<p>Convert a GDS file to a PLINK binary ped (BED) file.
</p>


<h3>Usage</h3>

<pre>
snpgdsGDS2BED(gdsobj, bed.fn, sample.id=NULL, snp.id=NULL, snpfirstdim=NULL,
    verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file; or characters, the file name of GDS</p>
</td></tr>
<tr valign="top"><td><code>bed.fn</code></td>
<td>
<p>the file name of output, without the filename extension &quot;.bed&quot;</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs;
if NULL, all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>snpfirstdim</code></td>
<td>
<p>if TRUE, genotypes are stored in the individual-major
mode, (i.e, list all SNPs for the first individual, and then list all
SNPs for the second individual, etc); if NULL, determine
automatically</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GDS &ndash; Genomic Data Structures, the extended file name used for storing
genetic data, and the file format used in the <a href="../../gdsfmt/html/gdsfmt-package.html">gdsfmt</a> package.
</p>
<p>BED &ndash; the PLINK binary ped format.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D,
Maller J, Sklar P, de Bakker PIW, Daly MJ &amp; Sham PC. 2007.
PLINK: a toolset for whole-genome association and population-based linkage
analysis. American Journal of Human Genetics, 81.
</p>
<p><a href="http://corearray.sourceforge.net/">http://corearray.sourceforge.net/</a>
</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsBED2GDS.html">snpgdsBED2GDS</a></code>, <code><a href="../../SNPRelate/help/snpgdsGDS2PED.html">snpgdsGDS2PED</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

snpset &lt;- snpgdsSelectSNP(genofile, missing.rate=0.95)
snpgdsGDS2BED(genofile, bed.fn="test", snp.id=snpset)

# close the genotype file
snpgdsClose(genofile)


# delete the temporary files
unlink(c("test.bed", "test.bim", "test.fam"), force=TRUE)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
