<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Individual Inbreeding Coefficients</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsIndInb {SNPRelate}"><tr><td>snpgdsIndInb {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Individual Inbreeding Coefficients
</h2>

<h3>Description</h3>

<p>To calculate individual inbreeding coefficients using SNP genotype data
</p>


<h3>Usage</h3>

<pre>
snpgdsIndInb(gdsobj, sample.id=NULL, snp.id=NULL,
    autosome.only=TRUE, remove.monosnp=TRUE, maf=NaN, missing.rate=NaN,
    method=c("mom.weir", "mom.visscher", "mle", "gcta1", "gcta2", "gcta3"),
    allele.freq=NULL, out.num.iter=TRUE, reltol=.Machine$double.eps^0.75,
    verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs;
if NULL, all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>autosome.only</code></td>
<td>
<p>if <code>TRUE</code>, use autosomal SNPs only; if it is a
numeric or character value, keep SNPs according to the specified
chromosome</p>
</td></tr>
<tr valign="top"><td><code>remove.monosnp</code></td>
<td>
<p>if TRUE, remove monomorphic SNPs</p>
</td></tr>
<tr valign="top"><td><code>maf</code></td>
<td>
<p>to use the SNPs with &quot;&gt;= maf&quot; only; if NaN, no MAF threshold</p>
</td></tr>
<tr valign="top"><td><code>missing.rate</code></td>
<td>
<p>to use the SNPs with &quot;&lt;= missing.rate&quot; only; if NaN,
no missing threshold</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>see details</p>
</td></tr>
<tr valign="top"><td><code>allele.freq</code></td>
<td>
<p>to specify the allele frequencies; if NULL, the allele
frequencies are estimated from the given samples</p>
</td></tr>
<tr valign="top"><td><code>out.num.iter</code></td>
<td>
<p>output the numbers of iterations</p>
</td></tr>
<tr valign="top"><td><code>reltol</code></td>
<td>
<p>relative convergence tolerance used in MLE; the algorithm
stops if it is unable to reduce the value of log likelihood by a
factor of $reltol * (abs(log likelihood with the initial parameters)
+ reltol)$ at a step.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>method</code> can be:
&quot;mom.weir&quot;: a modified Visscher's estimator, proposed by Bruce Weir;
&quot;mom.visscher&quot;: Visscher's estimator described in Yang et al. (2010);
&quot;mle&quot;: the maximum likelihood estimation;
&quot;gcta1&quot;: F^I in GCTA, avg [(g_i - 2p_i)^2 / (2*p_i*(1-p_i)) - 1];
&quot;gcta2&quot;: F^II in GCTA, avg [1 - g_i*(2 - g_i) / (2*p_i*(1-p_i))];
&quot;gcta3&quot;: F^III in GCTA, the same as &quot;mom.visscher&quot;,
avg [g_i^2 - (1 + 2p_i)*g_i + 2*p_i^2] / (2*p_i*(1-p_i)).
</p>


<h3>Value</h3>

<p>Return estimated inbreeding coefficient.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Yang J, Benyamin B, McEvoy BP, Gordon S, Henders AK, Nyholt DR, Madden PA,
Heath AC, Martin NG, Montgomery GW, Goddard ME, Visscher PM. 2010.
Common SNPs explain a large proportion of the heritability for human height.
Nat Genet. 42(7):565-9. Epub 2010 Jun 20.
</p>
<p>Yang, J., Lee, S. H., Goddard, M. E. &amp; Visscher, P. M.
GCTA: a tool for genome-wide complex trait analysis.
American journal of human genetics 88, 76-82 (2011).
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

rv &lt;- snpgdsIndInb(genofile, method="mom.visscher")
head(rv$inbreeding)
summary(rv$inbreeding)

# close the genotype file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
