<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Option settings: chromosome coding, etc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsOption {SNPRelate}"><tr><td>snpgdsOption {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Option settings: chromosome coding, etc
</h2>

<h3>Description</h3>

<p>Return an option list used by the SNPRelate package or a GDS file
</p>


<h3>Usage</h3>

<pre>
snpgdsOption(gdsobj=NULL, autosome.start=1L, autosome.end=22L, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>autosome.start</code></td>
<td>
<p>the starting index of autosome</p>
</td></tr>
<tr valign="top"><td><code>autosome.end</code></td>
<td>
<p>the ending index of autosome</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>optional arguments for new chromosome coding</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>Examples</h3>

<pre>
# define the new chromosomes 'Z' and 'W'
snpgdsOption(Z=27L, W=28L)


# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

snpgdsOption(genofile)

# close the genotype file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
