<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: To get a genotype matrix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsGetGeno {SNPRelate}"><tr><td>snpgdsGetGeno {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
To get a genotype matrix
</h2>

<h3>Description</h3>

<p>To get a genotype matrix from a specified GDS file
</p>


<h3>Usage</h3>

<pre>
snpgdsGetGeno(gdsobj, sample.id=NULL, snp.id=NULL, snpfirstdim=NA,
    .snpread=NA, with.id=FALSE, verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file; or characters to specify the file name of SNP GDS</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs;
if NULL, all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>snpfirstdim</code></td>
<td>
<p>if <code>TRUE</code>, genotypes are stored in the
individual-major mode, (i.e, list all SNPs for the first individual,
and then list all SNPs for the second individual, etc); <code>FALSE</code>
for snp-major mode; if <code>NA</code>, determine automatically</p>
</td></tr>
<tr valign="top"><td><code>.snpread</code></td>
<td>
<p>internal use</p>
</td></tr>
<tr valign="top"><td><code>with.id</code></td>
<td>
<p>if <code>TRUE</code>, return <code>sample.id</code> and <code>snp.id</code></p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns an integer matrix with values 0, 1, 2 or NA
representing the number of reference allele when <code>with.id=FALSE</code>;
or <code>list(genotype, sample.id, snp.id)</code> when <code>with.id=TRUE</code>. The
orders of sample and SNP IDs in the genotype matrix are actually consistent
with <code>sample.id</code> and <code>snp.id</code> in the GDS file, which may not be
as the same as the arguments <code>sampel.id</code> and <code>snp.id</code> specified
by users.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

set.seed(1000)
snpset &lt;- sample(read.gdsn(index.gdsn(genofile, "snp.id")), 1000)

mat1 &lt;- snpgdsGetGeno(genofile, snp.id=snpset, snpfirstdim=TRUE)
dim(mat1)
# 1000  279
table(c(mat1), exclude=NULL)

mat2 &lt;- snpgdsGetGeno(genofile, snp.id=snpset, snpfirstdim=FALSE)
dim(mat2)
# 279 1000
table(c(mat2), exclude=NULL)

identical(t(mat1), mat2)
# TRUE

# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
