<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Reformat a VCF file (R implementation)</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsVCF2GDS_R {SNPRelate}"><tr><td>snpgdsVCF2GDS_R {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Reformat a VCF file (R implementation)</h2>

<h3>Description</h3>

<p>Reformat a Variant Call Format (VCF) file
</p>


<h3>Usage</h3>

<pre>
snpgdsVCF2GDS_R(vcf.fn, out.fn, nblock=1024,
    method = c("biallelic.only", "copy.num.of.ref"),
    compress.annotation="LZMA_RA", snpfirstdim=FALSE, option = NULL,
    verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>vcf.fn</code></td>
<td>
<p>the file name of VCF format, <code>vcf.fn</code> can be a vector,
see details</p>
</td></tr>
<tr valign="top"><td><code>out.fn</code></td>
<td>
<p>the output gds file</p>
</td></tr>
<tr valign="top"><td><code>nblock</code></td>
<td>
<p>the buffer lines</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>either &quot;biallelic.only&quot; by default or &quot;copy.num.of.ref&quot;,
see details</p>
</td></tr>
<tr valign="top"><td><code>compress.annotation</code></td>
<td>
<p>the compression method for the GDS variables,
except &quot;genotype&quot;; optional values are defined in the function
<code>add.gdsn</code></p>
</td></tr>
<tr valign="top"><td><code>snpfirstdim</code></td>
<td>
<p>if TRUE, genotypes are stored in the individual-major
mode, (i.e, list all SNPs for the first individual, and then list all
SNPs for the second individual, etc)</p>
</td></tr>
<tr valign="top"><td><code>option</code></td>
<td>
<p><code>NULL</code> or an object from <code><a href="../../SNPRelate/help/snpgdsOption.html">snpgdsOption</a></code>,
see details</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GDS &ndash; Genomic Data Structures used for storing genetic array-oriented
data, and the file format used in the <a href="../../gdsfmt/html/gdsfmt-package.html">gdsfmt</a> package.
</p>
<p>VCF &ndash; The Variant Call Format (VCF), which is a generic format for storing
DNA polymorphism data such as SNPs, insertions, deletions and
structural variants, together with rich annotations.
</p>
<p>If there are more than one file name in <code>vcf.fn</code>, <code>snpgdsVCF2GDS</code>
will merge all dataset together once they all contain the same samples. It is
useful to combine genetic data if VCF data are divided by chromosomes.
</p>
<p><code>method = "biallelic.only"</code>: to exact bi-allelic and polymorhpic
SNP data (excluding monomorphic variants);
<code>method = "biallelic.only"</code>: to exact bi-allelic and polymorhpic SNP
data; <code>method = "copy.num.of.ref"</code>: to extract and store dosage (0, 1, 2)
of the reference allele for all variant sites, including bi-allelic SNPs,
multi-allelic SNPs, indels and structural variants.
</p>
<p>Haploid and triploid calls are allowed in the transfer, the variable
<code>snp.id</code> stores the original the row index of variants, and the variable
<code>snp.rs.id</code> stores the rs id.
</p>
<p>The user could use <code>option</code> to specify the range of code for autosomes.
For humans there are 22 autosomes (from 1 to 22), but dogs have 38 autosomes.
Note that the default settings are used for humans. The user could call
<code>option = snpgdsOption(autosome.end=38)</code> for importing the VCF file of dog.
It also allows defining new chromosome coding, e.g.,
<code>option = snpgdsOption(Z=27)</code>, then &quot;Z&quot; will be replaced by the number 27.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>The variant call format and VCFtools.
Danecek P, Auton A, Abecasis G, Albers CA, Banks E, DePristo MA,
Handsaker RE, Lunter G, Marth GT, Sherry ST, McVean G, Durbin R;
1000 Genomes Project Analysis Group.
Bioinformatics. 2011 Aug 1;27(15):2156-8. Epub 2011 Jun 7.
</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsVCF2GDS_R.html">snpgdsVCF2GDS_R</a></code>, <code><a href="../../SNPRelate/help/snpgdsOption.html">snpgdsOption</a></code>,
<code><a href="../../SNPRelate/help/snpgdsBED2GDS.html">snpgdsBED2GDS</a></code>
</p>


<h3>Examples</h3>

<pre>
# The VCF file
vcf.fn &lt;- system.file("extdata", "sequence.vcf", package="SNPRelate")
cat(readLines(vcf.fn), sep="\n")

snpgdsVCF2GDS_R(vcf.fn, "test1.gds", method="biallelic.only")
snpgdsSummary("test1.gds")

snpgdsVCF2GDS_R(vcf.fn, "test2.gds", method="biallelic.only")
snpgdsSummary("test2.gds")

snpgdsVCF2GDS_R(vcf.fn, "test3.gds", method="copy.num.of.ref")
snpgdsSummary("test3.gds")

snpgdsVCF2GDS_R(vcf.fn, "test4.gds", method="copy.num.of.ref")
snpgdsSummary("test4.gds")
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
