<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Individual inbreeding and relatedness estimation (beta...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsIndivBeta {SNPRelate}"><tr><td>snpgdsIndivBeta {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Individual inbreeding and relatedness estimation (beta estimator)
</h2>

<h3>Description</h3>

<p>Calculate individual inbreeding and relatedness estimation (beta estimator)
using SNP genotype data.
</p>


<h3>Usage</h3>

<pre>
snpgdsIndivBeta(gdsobj, sample.id=NULL, snp.id=NULL, autosome.only=TRUE,
    remove.monosnp=TRUE, maf=NaN, missing.rate=NaN, method=c("weighted"),
    inbreeding=TRUE, num.thread=1L, with.id=TRUE, useMatrix=FALSE, verbose=TRUE)
snpgdsIndivBetaRel(beta, beta_rel, verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs; if NULL,
all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>autosome.only</code></td>
<td>
<p>if <code>TRUE</code>, use autosomal SNPs only; if it is a
numeric or character value, keep SNPs according to the specified
chromosome</p>
</td></tr>
<tr valign="top"><td><code>remove.monosnp</code></td>
<td>
<p>if TRUE, remove monomorphic SNPs</p>
</td></tr>
<tr valign="top"><td><code>maf</code></td>
<td>
<p>to use the SNPs with &quot;&gt;= maf&quot; only; if NaN, no MAF threshold</p>
</td></tr>
<tr valign="top"><td><code>missing.rate</code></td>
<td>
<p>to use the SNPs with &quot;&lt;= missing.rate&quot; only; if NaN,
no missing threshold</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>&quot;weighted&quot; estimator</p>
</td></tr>
<tr valign="top"><td><code>inbreeding</code></td>
<td>
<p><code>TRUE</code>, the diagonal is a vector of inbreeding
coefficients; otherwise, individual variance estimates</p>
</td></tr>
<tr valign="top"><td><code>num.thread</code></td>
<td>
<p>the number of (CPU) cores used; if <code>NA</code>, detect
the number of cores automatically</p>
</td></tr>
<tr valign="top"><td><code>with.id</code></td>
<td>
<p>if <code>TRUE</code>, the returned value with <code>sample.id</code>
and <code>sample.id</code></p>
</td></tr>
<tr valign="top"><td><code>useMatrix</code></td>
<td>
<p>if <code>TRUE</code>, use <code>Matrix::dspMatrix</code> to store
the output square matrix to save memory</p>
</td></tr>
<tr valign="top"><td><code>beta</code></td>
<td>
<p>the object returned from <code>snpgdsIndivBeta()</code></p>
</td></tr>
<tr valign="top"><td><code>beta_rel</code></td>
<td>
<p>the beta-based matrix is generated relative to
<code>beta_rel</code></p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if <code>TRUE</code>, show information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list if <code>with.id = TRUE</code>:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>the sample ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>the SNP ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>inbreeding</code></td>
<td>
<p>a logical value; <code>TRUE</code>, the diagonal is a
vector of inbreeding coefficients; otherwise, individual variance
estimates</p>
</td></tr>
<tr valign="top"><td><code>beta</code></td>
<td>
<p>beta estimates</p>
</td></tr>
<tr valign="top"><td><code>avg_val</code></td>
<td>
<p>the average of M_B among all loci, it could be used to
calculate each M_ij</p>
</td></tr>
</table>
<p>If <code>with.id = FALSE</code>, this function returns the genetic relationship
matrix without sample and SNP IDs.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Weir BS, Zheng X. SNPs and SNVs in Forensic Science.
Forensic Science International: Genetics Supplement Series. 2015.
doi:10.1016/j.fsigss.2015.09.106
</p>
<p>Weir BS, Goudet J. A Unified Characterization of Population Structure and
Relatedness. Genetics. 2017 Aug;206(4):2085-2103.
doi: 10.1534/genetics.116.198424.
</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsGRM.html">snpgdsGRM</a></code>, <code><a href="../../SNPRelate/help/snpgdsIndInb.html">snpgdsIndInb</a></code>,
<code><a href="../../SNPRelate/help/snpgdsFst.html">snpgdsFst</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

b &lt;- snpgdsIndivBeta(genofile, inbreeding=FALSE)
b$beta[1:10, 1:10]

z &lt;- snpgdsIndivBetaRel(b, min(b$beta))

# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
