<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Get a common SNP list between/among SNP list objects</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsSNPListIntersect {SNPRelate}"><tr><td>snpgdsSNPListIntersect {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Get a common SNP list between/among SNP list objects
</h2>

<h3>Description</h3>

<p>Get a common SNP list by comparing their snp id, chromosome, positions
and allele frequency if needed.
</p>


<h3>Usage</h3>

<pre>
snpgdsSNPListIntersect(snplist1, snplist2, ..., method=c("position", "exact"),
    na.rm=TRUE, same.strand=FALSE, verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>snplist1</code></td>
<td>
<p>the SNP list object <code><a href="../../SNPRelate/help/snpgdsSNPListClass.html">snpgdsSNPListClass</a></code></p>
</td></tr>
<tr valign="top"><td><code>snplist2</code></td>
<td>
<p>the SNP list object <code><a href="../../SNPRelate/help/snpgdsSNPListClass.html">snpgdsSNPListClass</a></code></p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>the other SNP list objects</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p><code>"exact"</code>: matching by all snp.id, chromosomes, positions
and alleles; <code>"position"</code>: matching by chromosomes and positions</p>
</td></tr>
<tr valign="top"><td><code>na.rm</code></td>
<td>
<p>if TRUE, remove mismatched alleles</p>
</td></tr>
<tr valign="top"><td><code>same.strand</code></td>
<td>
<p>if TRUE, assuming the alleles on the same strand</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list of <code>snpgdsSNPListClass</code> including the following
components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>idx1</code></td>
<td>
<p>the indices of common SNPs in the first GDS file</p>
</td></tr>
<tr valign="top"><td><code>idx2</code></td>
<td>
<p>the indices of common SNPs in the second GDS file</p>
</td></tr>
<tr valign="top"><td><code>idx...</code></td>
<td>
</td></tr>
<tr valign="top"><td><code>idx<code>n</code></code></td>
<td>
<p>the indices of common SNPs in the n-th GDS file</p>
</td></tr>
<tr valign="top"><td><code>flag2</code></td>
<td>
<p>an integer vector, flip flag for each common SNP for the
second GDS file (assuming a value <code>v</code>):
<code>bitwAnd(v, 1)</code>: 0 &ndash; no flip of allele names,
1 &ndash; flip of allele names;
<code>bitwAnd(v, 2)</code>: 0 &ndash; on the same strand,
2 &ndash; on the different strands, comparing with the first GDS file;
<code>bitwAnd(v, 4)</code>: 0 &ndash; no strand ambiguity,
4 &ndash; ambiguous allele names, determined by allele frequencies;
<code>NA</code> &ndash; mismatched allele names
(there is no NA if <code>na.rm=TRUE</code>)</p>
</td></tr>
<tr valign="top"><td><code>flag...</code></td>
<td>
</td></tr>
<tr valign="top"><td><code>flag<code>n</code></code></td>
<td>
<p>flip flag for each common SNP for the n-th GDS file</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsSNPList.html">snpgdsSNPList</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

# to get a snp list object
snplist1 &lt;- snpgdsSNPList(genofile)
snplist2 &lt;- snpgdsSNPList(genofile)

# a common snp list, exactly matching
v &lt;- snpgdsSNPListIntersect(snplist1, snplist2)
names(v)
# "idx1" "idx2"

# a common snp list, matching by position
v &lt;- snpgdsSNPListIntersect(snplist1, snplist2, method="pos")
names(v)
# "idx1"  "idx2"  "flag2"

table(v$flag2, exclude=NULL)


# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
