<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Eigen-analysis on SNP genotype data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for snpgdsEIGMIX {SNPRelate}"><tr><td>snpgdsEIGMIX {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Eigen-analysis on SNP genotype data
</h2>

<h3>Description</h3>

<p>Eigen-analysis on IBD matrix based SNP genotypes.
</p>


<h3>Usage</h3>

<pre>
snpgdsEIGMIX(gdsobj, sample.id=NULL, snp.id=NULL, autosome.only=TRUE,
    remove.monosnp=TRUE, maf=NaN, missing.rate=NaN, num.thread=1L,
    eigen.cnt=32L, diagadj=TRUE, ibdmat=FALSE, verbose=TRUE)
## S3 method for class 'snpgdsEigMixClass'
plot(x, eig=c(1L,2L), ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="../../SNPRelate/help/SNPGDSFileClass.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs; if NULL,
all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>autosome.only</code></td>
<td>
<p>if <code>TRUE</code>, use autosomal SNPs only; if it is a
numeric or character value, keep SNPs according to the specified
chromosome</p>
</td></tr>
<tr valign="top"><td><code>remove.monosnp</code></td>
<td>
<p>if <code>TRUE</code>, remove monomorphic SNPs</p>
</td></tr>
<tr valign="top"><td><code>maf</code></td>
<td>
<p>to use the SNPs with &quot;&gt;= maf&quot; only; if NaN, no MAF threshold</p>
</td></tr>
<tr valign="top"><td><code>missing.rate</code></td>
<td>
<p>to use the SNPs with &quot;&lt;= missing.rate&quot; only; if NaN,
no missing threshold</p>
</td></tr>
<tr valign="top"><td><code>num.thread</code></td>
<td>
<p>the number of (CPU) cores used; if <code>NA</code>, detect
the number of cores automatically</p>
</td></tr>
<tr valign="top"><td><code>eigen.cnt</code></td>
<td>
<p>output the number of eigenvectors; if <code>eigen.cnt &lt; 0</code>,
returns all eigenvectors; if <code>eigen.cnt==0</code>, no eigen calculation</p>
</td></tr>
<tr valign="top"><td><code>diagadj</code></td>
<td>
<p><code>TRUE</code> for diagonal adjustment by default</p>
</td></tr>
<tr valign="top"><td><code>ibdmat</code></td>
<td>
<p>if <code>TRUE</code>, returns the IBD matrix</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if <code>TRUE</code>, show information</p>
</td></tr>
<tr valign="top"><td><code>x</code></td>
<td>
<p>a <code>snpgdsEigMixClass</code> object</p>
</td></tr>
<tr valign="top"><td><code>eig</code></td>
<td>
<p>indices of eigenvectors, like <code>1:2</code> or <code>1:4</code></p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>the arguments passed to or from other methods, like
<code>pch</code>, <code>col</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a <code>snpgdsEigMixClass</code> object, and it is a list:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>the sample ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>the SNP ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>eigenval</code></td>
<td>
<p>eigenvalues</p>
</td></tr>
<tr valign="top"><td><code>eigenvect</code></td>
<td>
<p>eigenvactors, &quot;# of samples&quot; x &quot;eigen.cnt&quot;</p>
</td></tr>
<tr valign="top"><td><code>afreq</code></td>
<td>
<p>allele frequencies</p>
</td></tr>
<tr valign="top"><td><code>ibd</code></td>
<td>
<p>the IBD matrix when <code>ibdmat=TRUE</code></p>
</td></tr>
<tr valign="top"><td><code>diagadj</code></td>
<td>
<p>the argument <code>diagadj</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Zheng X, Weir BS.
Eigenanalysis on SNP Data with an Interpretation of Identity by Descent.
Theoretical Population Biology. 2016 Feb;107:65-76.
doi: 10.1016/j.tpb.2015.09.004
</p>


<h3>See Also</h3>

<p><code><a href="../../SNPRelate/help/snpgdsAdmixProp.html">snpgdsAdmixProp</a></code>, <code><a href="../../SNPRelate/help/snpgdsAdmixPlot.html">snpgdsAdmixPlot</a></code>,
<code><a href="../../SNPRelate/help/snpgdsPCA.html">snpgdsPCA</a></code>, <code><a href="../../SNPRelate/help/snpgdsPCASNPLoading.html">snpgdsPCASNPLoading</a></code>, 
<code><a href="../../SNPRelate/help/snpgdsPCASampLoading.html">snpgdsPCASampLoading</a></code> 
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

# get population information
#   or pop_code &lt;- scan("pop.txt", what=character())
#   if it is stored in a text file "pop.txt"
pop_code &lt;- read.gdsn(index.gdsn(genofile, "sample.annot/pop.group"))

# get sample id
samp.id &lt;- read.gdsn(index.gdsn(genofile, "sample.id"))

# run eigen-analysis
RV &lt;- snpgdsEIGMIX(genofile)
RV

# eigenvalues
RV$eigenval

# make a data.frame
tab &lt;- data.frame(sample.id = samp.id, pop = factor(pop_code),
    EV1 = RV$eigenvect[,1],    # the first eigenvector
    EV2 = RV$eigenvect[,2],    # the second eigenvector
    stringsAsFactors = FALSE)
head(tab)

# draw
plot(tab$EV2, tab$EV1, col=as.integer(tab$pop),
    xlab="eigenvector 2", ylab="eigenvector 1")
legend("topleft", legend=levels(tab$pop), pch="o", col=1:4)


# define groups
groups &lt;- list(CEU = samp.id[pop_code == "CEU"],
    YRI = samp.id[pop_code == "YRI"],
    CHB = samp.id[is.element(pop_code, c("HCB", "JPT"))])

prop &lt;- snpgdsAdmixProp(RV, groups=groups)

# draw
plot(prop[, "YRI"], prop[, "CEU"], col=as.integer(tab$pop),
    xlab = "Admixture Proportion from YRI",
    ylab = "Admixture Proportion from CEU")
abline(v=0, col="gray25", lty=2)
abline(h=0, col="gray25", lty=2)
abline(a=1, b=-1, col="gray25", lty=2)
legend("topright", legend=levels(tab$pop), pch="o", col=1:4)


# close the genotype file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
