<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Parallel Computing Toolset for Genome-Wide Association...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for SNPRelate-package {SNPRelate}"><tr><td>SNPRelate-package {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Parallel Computing Toolset for Genome-Wide Association Studies</h2>

<h3>Description</h3>

<p>Genome-wide association studies are widely used to investigate
the genetic basis of diseases and traits, but they pose many computational
challenges. We developed SNPRelate (R package for multi-core symmetric
multiprocessing computer architectures) to accelerate two key computations
on SNP data: principal component analysis (PCA) and relatedness analysis
using identity-by-descent measures. The kernels of our algorithms are
written in C/C++ and highly optimized.
</p>


<h3>Details</h3>


<table summary="Rd table">
<tr>
 <td style="text-align: left;">
        Package: </td><td style="text-align: left;"> SNPRelate </td>
</tr>
<tr>
 <td style="text-align: left;">
        Type: </td><td style="text-align: left;"> Package </td>
</tr>
<tr>
 <td style="text-align: left;">
        License: </td><td style="text-align: left;"> GPL version 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
        Depends: </td><td style="text-align: left;"> gdsfmt (&gt;= 1.0.4) </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

<p>The genotypes stored in GDS format can be analyzed by the R functions in
SNPRelate, which utilize the multi-core feature of machine for a single
computer.
</p>
<p>Webpage: <a href="http://github.com/zhengxwen/SNPRelate">http://github.com/zhengxwen/SNPRelate</a>,
<a href="http://corearray.sourceforge.net/">http://corearray.sourceforge.net/</a>
</p>
<p>Tutorial: <a href="http://corearray.sourceforge.net/tutorials/SNPRelate/">http://corearray.sourceforge.net/tutorials/SNPRelate/</a>
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng <a href="mailto:zhengxwen@gmail.com">zhengxwen@gmail.com</a>
</p>


<h3>References</h3>

<p>Zheng X, Levine D, Shen J, Gogarten SM, Laurie C, Weir BS.
A High-performance Computing Toolset for Relatedness and Principal
Component Analysis of SNP Data.
Bioinformatics (2012); doi: 10.1093/bioinformatics/bts610
</p>


<h3>Examples</h3>

<pre>
####################################################################
# Convert the PLINK BED file to the GDS file
#

# PLINK BED files
bed.fn &lt;- system.file("extdata", "plinkhapmap.bed.gz", package="SNPRelate")
fam.fn &lt;- system.file("extdata", "plinkhapmap.fam.gz", package="SNPRelate")
bim.fn &lt;- system.file("extdata", "plinkhapmap.bim.gz", package="SNPRelate")

# convert
snpgdsBED2GDS(bed.fn, fam.fn, bim.fn, "HapMap.gds")


####################################################################
# Principal Component Analysis
#

# open
genofile &lt;- snpgdsOpen("HapMap.gds")

RV &lt;- snpgdsPCA(genofile)
plot(RV$eigenvect[,2], RV$eigenvect[,1], xlab="PC 2", ylab="PC 1",
    col=rgb(0,0,150, 50, maxColorValue=255), pch=19)

# close the file
snpgdsClose(genofile)


####################################################################
# Identity-By-Descent (IBD) Analysis
#

# open
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

RV &lt;- snpgdsIBDMoM(genofile)
flag &lt;- lower.tri(RV$k0)
plot(RV$k0[flag], RV$k1[flag], xlab="k0", ylab="k1",
    col=rgb(0,0,150, 50, maxColorValue=255), pch=19)
abline(1, -1, col="red", lty=4)

# close the file
snpgdsClose(genofile)


####################################################################
# Identity-By-State (IBS) Analysis
#

# open
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

RV &lt;- snpgdsIBS(genofile)
m &lt;- 1 - RV$ibs
colnames(m) &lt;- rownames(m) &lt;- RV$sample.id
GeneticDistance &lt;- as.dist(m[1:45, 1:45])
HC &lt;- hclust(GeneticDistance, "ave")
plot(HC)

# close the file
snpgdsClose(genofile)


####################################################################
# Linkage Disequilibrium (LD) Analysis
#

# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

snpset &lt;- read.gdsn(index.gdsn(genofile, "snp.id"))[1:200]
L1 &lt;- snpgdsLDMat(genofile, snp.id=snpset, method="composite", slide=-1)

# plot
image(abs(L1$LD), col=terrain.colors(64))

# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.28.0 <a href="00Index.html">Index</a>]</div>
</div></body></html>
