<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: plot association with LD and annotation at a given region</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for IntRegionalPlot {IntAssoPlot}"><tr><td>IntRegionalPlot {IntAssoPlot}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>plot association with LD and annotation at a given region</h2>

<h3>Description</h3>

<p>This function plot the association with
linkage disequiblism and annotation at the level of a region.
</p>


<h3>Usage</h3>

<pre>
IntRegionalPlot(
  chr,
  left,
  right,
  gtf,
  association,
  hapmap,
  hapmap_ld = NULL,
  slide_length = -1,
  threadN = 1,
  ldstatistics = "rsquare",
  leadsnp = NULL,
  threshold = NULL,
  link2gene = NULL,
  triangleLD = TRUE,
  link2LD = NULL,
  leadsnpLD = TRUE,
  label_gene_name = FALSE,
  colour02 = "gray",
  colour04 = "cyan",
  colour06 = "green",
  colour08 = "yellow",
  colour10 = "red",
  leadsnp_shape = 23,
  leadsnp_colour = "black",
  leadsnp_fill = "purple",
  leadsnp_size = 1.5,
  marker2highlight = NULL,
  marker2label = NULL,
  marker2label_angle = 60,
  marker2label_size = 1
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>chr</code></td>
<td>
<p>the chromosome, required.</p>
</td></tr>
<tr valign="top"><td><code>left</code></td>
<td>
<p>the left border of the region, required.</p>
</td></tr>
<tr valign="top"><td><code>right</code></td>
<td>
<p>the right border of the region, required.</p>
</td></tr>
<tr valign="top"><td><code>gtf</code></td>
<td>
<p>the annotation file, required.</p>
</td></tr>
<tr valign="top"><td><code>association</code></td>
<td>
<p>the association table, required.</p>
</td></tr>
<tr valign="top"><td><code>hapmap</code></td>
<td>
<p>the genotype file for computing leadsnpLD in the format of hapmap, required.</p>
</td></tr>
<tr valign="top"><td><code>hapmap_ld</code></td>
<td>
<p>the genotype file for computing trangleLD in the format of hapmap, not required. If hapmap_ld was not provided, hapmap would be used.</p>
</td></tr>
<tr valign="top"><td><code>slide_length</code></td>
<td>
<p>the sliding window length for computing LD, default -1.</p>
</td></tr>
<tr valign="top"><td><code>threadN</code></td>
<td>
<p>the number of (CPU) cores used for computing LD, default 1.</p>
</td></tr>
<tr valign="top"><td><code>ldstatistics</code></td>
<td>
<p>the statistics used for computing LD, default rsquare, and the optional is dprime.</p>
</td></tr>
<tr valign="top"><td><code>leadsnp</code></td>
<td>
<p>snp name provided by user</p>
</td></tr>
<tr valign="top"><td><code>threshold</code></td>
<td>
<p>the significant level of the assocition, default NULL.</p>
</td></tr>
<tr valign="top"><td><code>link2gene</code></td>
<td>
<p>a dataframe speicify markers to be linked from GWAS to genic structure, default NULL. When link2gene is 'NULL', locis that passed the threshold will be linked. Please see help('marker2link').</p>
</td></tr>
<tr valign="top"><td><code>triangleLD</code></td>
<td>
<p>show LD in the format lile triangle, default TRUE.</p>
</td></tr>
<tr valign="top"><td><code>link2LD</code></td>
<td>
<p>a dataframe speicify markers to be linked from genic structure to LD matrix, default NULL. When link2gene is 'NULL', locis that passed the threshold will be linked. Please see help('marker2link').</p>
</td></tr>
<tr valign="top"><td><code>leadsnpLD</code></td>
<td>
<p>show LD of the locis when compared with the most significant loci, default TRUE.</p>
</td></tr>
<tr valign="top"><td><code>label_gene_name</code></td>
<td>
<p>label the name of gene within the region, default FALSE.</p>
</td></tr>
<tr valign="top"><td><code>colour02</code></td>
<td>
<p>the colour of LD statistics ranged between 0.0 and 0.2, default gray.</p>
</td></tr>
<tr valign="top"><td><code>colour04</code></td>
<td>
<p>the colour of LD statistics ranged between 0.2 and 0.4, default cyan.</p>
</td></tr>
<tr valign="top"><td><code>colour06</code></td>
<td>
<p>the colour of LD statistics ranged between 0.4 and 0.6, default green.</p>
</td></tr>
<tr valign="top"><td><code>colour08</code></td>
<td>
<p>the colour of LD statistics ranged between 0.6 and 0.8, default yellow.</p>
</td></tr>
<tr valign="top"><td><code>colour10</code></td>
<td>
<p>the colour of LD statistics ranged between 0.8 and 1.0, default red.</p>
</td></tr>
<tr valign="top"><td><code>leadsnp_shape</code></td>
<td>
<p>the shape of leadsnp, default 23. For others, please see help('points').</p>
</td></tr>
<tr valign="top"><td><code>leadsnp_colour</code></td>
<td>
<p>the shape of the point of leadsnp, default black. For others, please see help('points').</p>
</td></tr>
<tr valign="top"><td><code>leadsnp_fill</code></td>
<td>
<p>the filled colour of the point of leadsnp, default purple. For others, please see help('points').</p>
</td></tr>
<tr valign="top"><td><code>leadsnp_size</code></td>
<td>
<p>the size of of the point of leadsnp, default 1.5. For others, please see help('points').</p>
</td></tr>
<tr valign="top"><td><code>marker2highlight</code></td>
<td>
<p>a dataframe speicify markers to be showed by colour,shape,fill,size, default NULL. Please see help('marker2highlight').</p>
</td></tr>
<tr valign="top"><td><code>marker2label</code></td>
<td>
<p>a dataframe speicify markers to be labeled, default NULL. Please see help('marker2link')</p>
</td></tr>
<tr valign="top"><td><code>marker2label_angle</code></td>
<td>
<p>angel of labeled text, default 60.</p>
</td></tr>
<tr valign="top"><td><code>marker2label_size</code></td>
<td>
<p>size of labeled text, default 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot
</p>


<h3>Author(s)</h3>

<p>Hongwei Wang &lt;<a href="mailto:whweve@163.com">whweve@163.com</a>&gt;
</p>


<h3>Examples</h3>

<pre>
data(gtf)
data(association)
data(hapmap_am368)
data(hapmap2)
IntRegionalPlot(chr=9,left=94178074-200000,right=94178074+200000,
gtf=gtf,association=association,hapmap=hapmap_am368,
hapmap_ld=hapmap_am368,threshold=5,leadsnp_size=2)
</pre>

<hr /><div style="text-align: center;">[Package <em>IntAssoPlot</em> version 0.99.21 <a href="00Index.html">Index</a>]</div>
</div></body></html>
