<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: To Get Some Protein Statistics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for AAstat {seqinr}"><tr><td>AAstat {seqinr}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2> To Get Some Protein Statistics </h2>

<h3>Description</h3>

<p>Returns simple protein sequence information including the number of residues,
the percentage physico-chemical classes and the theoretical isoelectric point.
The functions ignore ambiguous amino acids (e.g. &quot;B&quot;, &quot;Z&quot;, &quot;X&quot;, &quot;J&quot;).
</p>


<h3>Usage</h3>

<pre>
AAstat(seq, plot = TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>seq</code></td>
<td>
<p> a protein sequence as a vector of upper-case chars </p>
</td></tr>
<tr valign="top"><td><code>plot</code></td>
<td>
<p> if <code>TRUE</code>, plots the presence of residues splited by
physico-chemical classes along the sequence. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the three following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>Compo</code></td>
<td>
<p>A factor giving the amino acid counts.</p>
</td></tr>
<tr valign="top"><td><code>Prop</code></td>
<td>
<p>A list giving the percentage of each physico-chemical classes
(Tiny, Small, Aliphatic, Aromatic, Non-polar, Polar, Charged, Positive, Negative).</p>
</td></tr>
<tr valign="top"><td><code>Pi</code></td>
<td>
<p>The theoretical isoelectric point</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code><a href="../../seqinr/help/computePI.html">computePI</a></code>, <code><a href="../../seqinr/help/SEQINR.UTIL.html">SEQINR.UTIL</a></code>, <code><a href="../../seqinr/help/SeqFastaAA.html">SeqFastaAA</a></code></p>


<h3>Examples</h3>

<pre>
  seqAA &lt;- read.fasta(file = system.file("sequences/seqAA.fasta", package = "seqinr"),
   seqtype = "AA")
  AAstat(seqAA[[1]])
</pre>

<hr /><div style="text-align: center;">[Package <em>seqinr</em> version 4.2-36 <a href="00Index.html">Index</a>]</div>
</div></body></html>
