<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: To Compute the Theoretical Isoelectric Point</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for computePI {seqinr}"><tr><td>computePI {seqinr}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>To Compute the Theoretical Isoelectric Point</h2>

<h3>Description</h3>

<p>This function calculates the theoretical isoelectric point of a protein. Isoelectric point is the pH at which the protein has a neutral charge.
This estimate does not account for the post-translational modifications.
</p>


<h3>Usage</h3>

<pre>
computePI(seq)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>seq</code></td>
<td>
<p> Protein sequence as a vector of single chars in upper case</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The theoretical isoelectric point (pI) as a numerical vector of length one.
</p>


<h3>Note</h3>

 
<p>Protein pI is calculated using pK values of amino acids described in  Bjellqvist et al. See also <code>SEQINR.UTIL</code> for more details.
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

 
<p>The algorithm is the same as the one which is implemented at the following url:  
<a href="https://web.expasy.org/compute_pi/pi_tool-doc.html">https://web.expasy.org/compute_pi/pi_tool-doc.html</a> but with many trials
in case of convergence failure of the non-linear regression procedure.
<code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

<p><code><a href="../../seqinr/help/SEQINR.UTIL.html">SEQINR.UTIL</a></code> </p>


<h3>Examples</h3>

<pre>
#
# Simple sanity check with all 20 amino-acids in one-letter code alphabetical order:
#
prot &lt;- s2c("ACDEFGHIKLMNPQRSTVWY")
stopifnot(all.equal(computePI(prot), 6.78454))
#
# Read a protein sequence in a FASTA file and then compute its pI :
#
myProts &lt;- read.fasta(file = system.file("sequences/seqAA.fasta",
 package = "seqinr"), seqtype = "AA")
computePI(myProts[[1]]) # Should be 8.534902
</pre>

<hr /><div style="text-align: center;">[Package <em>seqinr</em> version 4.2-36 <a href="00Index.html">Index</a>]</div>
</div></body></html>
