<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Pairwise Distances from Aligned Protein or DNA/RNA Sequences</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for dist.alignment {seqinr}"><tr><td>dist.alignment {seqinr}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2> Pairwise Distances from Aligned Protein or DNA/RNA Sequences </h2>

<h3>Description</h3>

<p>These functions compute a matrix of pairwise distances from aligned sequences
using similarity (Fitch matrix, for protein sequences only) or identity matrix
(for protein and DNA sequences).
The resulting matrix contains the squared root of the pairwise distances.
For example, if identity between 2 sequences is 80
the squared root of (1.0 - 0.8) i.e. 0.4472136.
Note: seqinr::dist.alignment is the square root version of ape::dist.gene (and
not ape::dist.dna).
</p>


<h3>Usage</h3>

<pre>
dist.alignment(x, matrix = c("identity", "similarity"),gap)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>x</code></td>
<td>
<p>an object of class <code>alignment</code>, as returned by
<code>read.alignment</code> for instance</p>
</td></tr>
<tr valign="top"><td><code>matrix</code></td>
<td>
<p>the matrix distance to be used, partial matching allowed </p>
</td></tr>
<tr valign="top"><td><code>gap</code></td>
<td>
<p>-optional- logical, with identity matrix, if set to <code>TRUE</code>, gaps will be counted in the identity measure </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The distance matrix, object of class <code>dist</code>, computed by using the
specified distance measure.
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

<p>The reference for the similarity matrix is :<br />
Fitch, W.M. (1966) An improved method of testing for evolutionary homology.
<em>J. Mol. Biol.</em>, <b>16</b>:9-16.<br />
</p>
<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

  <p><code><a href="../../seqinr/help/read.alignment.html">read.alignment</a> </code> </p>


<h3>Examples</h3>

<pre>
 myseqs &lt;- read.alignment(file = system.file("sequences/test.mase",
 package = "seqinr"), format = "mase")
 dist.alignment(myseqs, matrix = "identity" )
 as.matrix(dist.alignment(myseqs, matrix = "identity" ))
</pre>

<hr /><div style="text-align: center;">[Package <em>seqinr</em> version 4.2-36 <a href="00Index.html">Index</a>]</div>
</div></body></html>
