<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Graphical representation for rearranged nucleotide skews in...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for draw.rearranged.oriloc {seqinr}"><tr><td>draw.rearranged.oriloc {seqinr}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Graphical representation for rearranged nucleotide skews in
prokaryotic chromosomes.</h2>

<h3>Description</h3>

<p>Graphical representation for rearranged nucleotide skews in
prokaryotic chromosomes.</p>


<h3>Usage</h3>

<pre>draw.rearranged.oriloc(rearr.ori, breaks.gcfw = NA,
 breaks.gcrev = NA, breaks.atfw = NA, breaks.atrev = NA)</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>rearr.ori</code></td>
<td>
<p>A data frame obtained with the <code>rearranged.oriloc</code>
function. </p>
</td></tr>
<tr valign="top"><td><code>breaks.gcfw</code></td>
<td>
<p>The coordinates of the breakpoints in the GC-skew,
for forward transcribed protein coding sequences. These coordinates
can be obtained with the <code>extract.breakpoints</code> function. </p>
</td></tr>
<tr valign="top"><td><code>breaks.gcrev</code></td>
<td>
<p>The coordinates of the breakpoints in the GC-skew,
for reverse transcribed protein coding sequences. These coordinates
can be obtained with the <code>extract.breakpoints</code> function. </p>
</td></tr>
<tr valign="top"><td><code>breaks.atfw</code></td>
<td>
<p>The coordinates of the breakpoints in the AT-skew,
for forward transcribed protein coding sequences. These coordinates
can be obtained with the <code>extract.breakpoints</code> function. </p>
</td></tr>
<tr valign="top"><td><code>breaks.atrev</code></td>
<td>
<p>The coordinates of the breakpoints in the AT-skew,
for reverse transcribed protein coding sequences. These coordinates
can be obtained with the <code>extract.breakpoints</code> function. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.R. Lobry, A. Necşulea</p>


<h3>References</h3>

<p>Necşulea, A. and Lobry, J.R. (2007) A New Method for Assessing the 
Effect of Replication on DNA Base Composition Asymmetry.
<em>Molecular Biology and Evolution</em>, <b>24</b>:2169-2179.
</p>


<h3>See Also</h3>

 <p><code><a href="../../seqinr/help/rearranged.oriloc.html">rearranged.oriloc</a></code>,
<code><a href="../../seqinr/help/extract.breakpoints.html">extract.breakpoints</a></code>  </p>


<h3>Examples</h3>

<pre>

## Not run: 
### Example for Chlamydia trachomatis ####

### Rearrange the chromosome and compute the nucleotide skews ###

#r.ori &lt;- rearranged.oriloc(seq.fasta = system.file("sequences/ct.fasta.gz", package = "seqinr"),
#    g2.coord = system.file("sequences/ct.coord", package = "seqinr"))

r.ori &lt;- rearranged.oriloc(seq.fasta = system.file("sequences/ct.fasta.gz", package = "seqinr"),
    g2.coord = system.file("sequences/ct.coord", package = "seqinr"))



### Extract the breakpoints for the rearranged nucleotide skews ###

breaks &lt;- extract.breakpoints(r.ori, type = c("gcfw", "gcrev"),
 nbreaks = c(2, 2), gridsize = 50, it.max = 100)

### Draw the rearranged nucleotide skews and  ###
### place the position of the breakpoints on the graphics ###

draw.rearranged.oriloc(r.ori, breaks.gcfw = breaks$gcfw$breaks,
 breaks.gcrev = breaks$gcrev$breaks)
## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>seqinr</em> version 4.2-36 <a href="00Index.html">Index</a>]</div>
</div></body></html>
