<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Extraction of breakpoint positions on the rearranged...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for extract.breakpoints {seqinr}"><tr><td>extract.breakpoints {seqinr}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Extraction of breakpoint positions on the
rearranged nucleotide skews. </h2>

<h3>Description</h3>

<p>Extraction of breakpoint positions on the
rearranged nucleotide skews. 
</p>


<h3>Usage</h3>

<pre>extract.breakpoints(rearr.ori,
type = c("atfw", "atrev", "gcfw", "gcrev"),
 nbreaks, gridsize = 100, it.max = 500)</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>rearr.ori</code></td>
<td>
<p>A data frame obtained with the <code>rearranged.oriloc</code>
function.  </p>
</td></tr>
<tr valign="top"><td><code>type</code></td>
<td>
<p>The type of skew for which to extract the breakpoints; must
be a subset of <code>c("atfw","atrev","gcfw","gcrev")</code>.</p>
</td></tr>
<tr valign="top"><td><code>nbreaks</code></td>
<td>
<p>The number of breakpoints to extract for each type of
skew. Provide a vector of the same length as <code>type</code>.</p>
</td></tr>
<tr valign="top"><td><code>gridsize</code></td>
<td>
<p>To make sure that the best breakpoints are found, and to
avoid finding only a local extremum of the likelihood and residual sum
of square functions, a grid search is performed. The search for
breakpoints is repeated <code>gridsize</code> times, with different starting
values for the breakpoints. </p>
</td></tr>
<tr valign="top"><td><code>it.max</code></td>
<td>
<p>The maximum number of iterations to be performed when
searching for the breakpoints. This argument corresponds to the
<code>it.max</code> argument in <code>segmented</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method uses the <code>segmented</code> function in the <code>segmented</code>
package to extract the breakpoints positions in the rearranged
nucleotide skews obtained with the <code>rearranged.oriloc</code> function.
To make sure that the best breakpoints are found, and to
avoid finding only a local extremum of the likelihood and residual sum
of square functions, a grid search is performed. The search for
breakpoints is repeated <code>gridsize</code> times, with different starting
values for the breakpoints. 
</p>


<h3>Value</h3>

<p>This function returns a list, with as many elements as the <code>type</code>
argument (for example <code>$gcfw</code> will contain the results for the
rearranged GC-skew, for forward-encoded genes). Each element of this list is also a list, containing the
following information: in  <code>$breaks</code> the position of the breakpoints on the
rearranged chromosome; in  <code>$slopes.left</code> the slopes of the
segments on the left side of each breakpoint; in  <code>$slopes.right</code> the slopes of the
segments on the right side of each breakpoint; in <code>$real.coord</code>,
the coordinates of the breakpoints on the real chromosome (before rearrangement).
</p>


<h3>Author(s)</h3>

<p>A. Necşulea</p>


<h3>References</h3>

<p><code>citation("segmented")</code>
</p>
<p>Necşulea, A. and Lobry, J.R. (in prep) A novel method for assessing
the effect of replication on DNA base composition asymmetry.
<em>Molecular Biology and Evolution</em>,<b>24</b>:2169-2179.
</p>


<h3>See Also</h3>

 <p><code><a href="../../seqinr/help/oriloc.html">oriloc</a></code>, <code><a href="../../seqinr/help/draw.rearranged.oriloc.html">draw.rearranged.oriloc</a></code>,
<code><a href="../../seqinr/help/rearranged.oriloc.html">rearranged.oriloc</a></code>  </p>


<h3>Examples</h3>

<pre>

### Example for Chlamydia trachomatis ####

### Rearrange the chromosome and compute the nucleotide skews ###

## Not run: r.ori &lt;- rearranged.oriloc(seq.fasta = system.file("sequences/ct.fasta.gz", package = "seqinr"),
    g2.coord = system.file("sequences/ct.coord",package = "seqinr"))
## End(Not run)

### Extract the breakpoints for the rearranged nucleotide skews ###


## Not run: breaks &lt;- extract.breakpoints(r.ori,type = c("gcfw", "gcrev"),
 nbreaks = c(2, 2), gridsize = 50, it.max = 100)
## End(Not run)


### Draw the rearranged nucleotide skews and ###
### place the position of the breakpoints on the graphics ###

## Not run: draw.rearranged.oriloc(r.ori, breaks.gcfw = breaks$gcfw$breaks,
 breaks.gcrev = breaks$gcrev$breaks)
## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>seqinr</em> version 4.2-36 <a href="00Index.html">Index</a>]</div>
</div></body></html>
